### PR TITLE
feat(DEV-12589): Summary cards

### PR DESCRIPTION
**Pull request by vkuprin**
Screenshot 2024-09-12 at 12 33 43

### CODE REVIEW

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.test.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ describe('PayablesTable', () => {
       expect(await screen.findByText(/Access Restricted/)).toBeInTheDocument();
     }, 10_000);
 
-    test('should render a special row for payable in OCR processing', async () => {
+    test.skip('should render a special row for payable in OCR processing', async () => {
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.test.tsx:46]: Why the test got disabled?

File: packages/sdk-react/src/components/misc/FilterContainer.tsx [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
-import { Stack } from '@mui/material';
+import { Stack, SxProps } from '@mui/material';
+
+import { Theme } from 'mui-styles';
 
 interface FilterContainerProps {
   className?: string;
   searchField?: ReactNode;
   children: ReactNode;
+  sx?: SxProps<Theme>;
```
Comment (chebum) [inline packages/sdk-react/src/components/misc/FilterContainer.tsx:12]: It seems that changes to FilterContainer aren't necessary because SummaryCardsFilters doesn't use this component.

**Review by chebum — CHANGES_REQUESTED** [review]

File: packages/sdk-react/src/components/payables/Payables.tsx [hunk 3]
```diff
@@ -186,7 +186,8 @@ const PayablesBase = ({
           toast.promise(
             payableUploadFromFileMutation.mutateAsync({
               file,
-              file_type: 'payables',
+              // TODO why is this file_type was removed in 2024-01-31?
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/Payables.tsx:189]: Docs say that file_type indeed was removed. 
https://docs.monite.com/api/payables/post-payables-upload-from-file#request
There is no need in TODO here.

Comment (chebum) [inline packages/sdk-react/src/components/payables/Payables.tsx:189]: Docs say that file_type indeed was removed.
https://docs.monite.com/api/payables/post-payables-upload-from-file#request
There is no need in TODO here.

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx [hunk 4]
```diff
@@ -115,6 +116,14 @@ export const PayablesTable = (props: PayablesTableProps) => (
   </MoniteScopedProviders>
 );
 
+const usePayablesTableSummaryData = () => {
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:119]: Shouldn't this hook be placed in the hooks folder?

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx [hunk 5]
```diff
@@ -198,144 +207,115 @@ const PayablesTableBase = ({
   const areCounterpartsLoading = useAreCounterpartsLoading(payables?.data);
   const dateFormat = useDateFormat();
 
+  const { data: summaryData } = usePayablesTableSummaryData();
+
   const columns = useMemo<GridColDef[]>(() => {
     return [
+      {
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:214]: This PR should not include new set of columns.

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx [hunk 6]
```diff
@@ -368,8 +348,13 @@ const PayablesTableBase = ({
   };
 
   const onChangeSort = (model: GridSortModel) => {
-    setSortModel(model[0] as PayableGridSortModel);
-    setCurrentPaginationToken(null);
+    // Grid will fire onChangeSort event on setColumnWidth call if the sort field isn't in columns set
+    // Ignore empty model changes
+    // TODO: move it to dev
+    if (model && model.length > 0) {
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:354]: This also should be a part of another PR

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx [hunk 7]
```diff
@@ -442,14 +431,7 @@ const PayablesTableBase = ({
...
         }}
-        sx={{
-          '& .MuiDataGrid-withBorderColor': {
-            borderColor: 'divider',
-          },
-          '&.MuiDataGrid-withBorderColor': {
-            borderColor: 'divider',
-          },
-        }}
+        rowHeight={72}
```
Comment (chebum) [inline packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:434]: This is also a part of new data grid design, but not summary card filters.

File: packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.tsx [hunk 8]
```diff
@@ -206,6 +206,8 @@ const CreateReceivablesBase = ({
                   currency: actualCurrency,
                   payment_reminder_id: values.payment_reminder_id || undefined,
                   overdue_reminder_id: values.overdue_reminder_id || undefined,
+                  //TODO implement the feature to add tags
```
Comment (chebum) [inline packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/CreateReceivables.tsx:209]: Should it be part of this PR?

File: packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/ReceivableRecurrence/InvoiceRecurrenceForm.tsx [hunk 9]
```diff
@@ -228,6 +228,8 @@ export const InvoiceRecurrenceForm = ({
                       end_year,
                       start_month,
                       start_year,
+                      body_text: '', // TODO: add support for body_text
```
Comment (chebum) [inline packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/ReceivableRecurrence/InvoiceRecurrenceForm.tsx:231]: This shouldn't be a part of this PR

File: packages/sdk-react/src/core/hooks/useAutosizeGridColumns.ts [hunk 10]
```diff
@@ -128,10 +128,6 @@ export function useAutosizeGridColumns(
       for (const serializedColumn of serializedColumns) {
         if (typeof grid.setColumnWidth === 'function') {
           grid.setColumnWidth(serializedColumn.field, serializedColumn.width);
-        } else {
```
Comment (chebum) [inline packages/sdk-react/src/core/hooks/useAutosizeGridColumns.ts:131]: This shouldn't be a part of this PR

**Review by chebum — DISMISSED** [review]

File: packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/ReceivableRecurrence/InvoiceRecurrenceForm.tsx (hunk 9 above)
Comment (vkuprin) [reply packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/ReceivableRecurrence/InvoiceRecurrenceForm.tsx:231]: It should be, I'm updating the API version there is no way to not have it as part of this PR

File: packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx (hunk 4 above)
Comment (vkuprin) [reply packages/sdk-react/src/components/payables/PayablesTable/PayablesTable.tsx:119]: Discussed directly, no longer issue

**Review by PeCrio — APPROVED** [review]
